eng/tools/typespec-validation/src/rules/sdk-tspconfig-validation.ts

```diff
@@ -340,43 +340,57 @@ export class TspConfigTsMlcDpPackageNameMatchPatternSubRule extends TspconfigEmi
     return skipForRestLevelClientOrManagementPlaneInTsEmitter(config, folder);
   }
 }
-// ----- Go data plane sub rules -----
-export class TspConfigGoDpServiceDirMatchPatternSubRule extends TspconfigEmitterOptionsSubRuleBase {
+
+// ----- Go common sub rules -----
+export class TspConfigGoModuleMatchPatternSubRule extends TspconfigEmitterOptionsSubRuleBase {
   constructor() {
-    super("@azure-tools/typespec-go", "service-dir", new RegExp(/^sdk\/.*$/));
+    super(
+      "@azure-tools/typespec-go",
+      "module",
+      new RegExp(/^github.com\/Azure\/azure-sdk-for-go\/.*$/),
+    );
   }
-  protected skip(_: any, folder: string) {
-    return skipForManagementPlane(folder);
+  protected validate(config: any): RuleResult {
+    let module = config?.options?.[this.emitterName]?.["module"];
+    if (module === undefined) return { success: true };
+    return super.validate(config);
   }
 }
 
-export class TspConfigGoDpPackageDirectoryMatchPatternSubRule extends TspconfigEmitterOptionsSubRuleBase {
+// ----- Go data plane sub rules -----
+export class TspConfigGoDpServiceDirMatchPatternSubRule extends TspconfigEmitterOptionsSubRuleBase {
   constructor() {
-    super("@azure-tools/typespec-go", "package-dir", new RegExp(/^az.*$/));
+    super("@azure-tools/typespec-go", "service-dir", new RegExp(/^(\{output-dir\}\/)?sdk\/.*$/));
```

Comment:
If we want to enforce putting `output-dir` in the `service-dir` for Go, make sure you fix all the violations in the 'main' branch.

Reply:
This is a potential violation example of having `output-dir` in the `emitter-output-dir`.
https://github.com/Azure/azure-rest-api-specs/pull/37018/files